File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/consumer/custom/annotation/ProductClient.java
package io.micronaut.rabbitmq.docs.consumer.custom.annotation;

import io.micronaut.context.annotation.Requires;
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;

@Requires(property = "spec.name", value = "DeliveryTagSpec")
@RabbitClient
public interface ProductClient {

    @Binding("product")
    void send(byte[] data);
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/consumer/custom/annotation/ProductClient.kt
package io.micronaut.configuration.rabbitmq.docs.consumer.custom.annotation

import io.micronaut.configuration.rabbitmq.annotation.Binding
import io.micronaut.configuration.rabbitmq.annotation.RabbitClient
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "DeliveryTagSpec")
@RabbitClient
interface ProductClient {

    @Binding("product")
    fun send(data: ByteArray)
}